providers/trello.js

```diff
@@ -0,0 +1,733 @@
+    }
+
+    _resolveFullCommentURL(card, commentID){
+        return this._resolveFullCardURL(card) + '#comment-' + commentID;
+    }
+
+    _resolveCardURL(id){
+        return this.baseLink + 'c/' + id;
+    }
+
+    _resolveBoardURL(id){
+        return this.baseLink + 'b/' + id;
+    }
+
+    _resolveCommentURL(cardID, commentID){
+        return this._resolveCardURL + '#comment-' + commentID;
+    }
+
+    _resolveGenericURL(id){
+        return this.baseLink + id;
+    }
+
+    _formatAttachment(attachment, embed){
+        if(attachment.previewUrl != null){
+            embed.image = {url: attachment.previewUrl};
+        }
+        if(attachment.url != null){
+            if(attachment.name !== attachment.url){
+                embed.fields = [{
+                    name: attachment.name,
+                    value: attachment.url
+                }];
+            } else {
+                embed.description = attachment.url; 
+            }
+        } else {
+            embed.description = attachment.name;
+        }
+    }
+
+    _formatLabel(text, value, embed){
+        if(value && this.defTrelloColors[value] != null){
+            this.payload.setEmbedColor(this.defTrelloColors[value]);
+        } else {
+            this.payload.setEmbedColor(this.defTrelloColors.nocolor);
+        }
+        if(text){
+            embed.description = '`' + text + '`';
+        }
+    }
+
+    _resolveUser(){
+        const memberCreator = this.req.body.action.memberCreator;
+        return {
+            name: memberCreator.fullName,
+            icon_url: this.avatarurl + memberCreator.avatarHash + '/170.png',
+            url: this.baseLink + memberCreator.username
+        };
+    }
+
+    _preparePayload(){
+        this.action = this.req.body.action;
+        this.model = this.req.body.model;
+
+        console.info(this.body);
+        console.info(this.action.data);
+
+        //Use the background color of the board if applicable. Otherwise, use the default trello color.
+        if(this.model.prefs != null && this.model.prefs.background != null && this.defTrelloColors[this.model.prefs.background] != null){
+            this.payload.setEmbedColor(this.defTrelloColors[this.model.prefs.background]);
+        } else {
+            this.payload.setEmbedColor(this.defTrelloColors.trello);
+        }
+
+        return {
+            author: this._resolveUser()
+        };
+    }
+
+    // Webhook Type Responses
+
+    async addAttachmentToCard(){
+        let embed = this._preparePayload();
+        embed.title = '[' + this.action.data.board.name + '] Added Attachment to "' + this.action.data.card.name + '"';
+        embed.url = this._resolveFullCardURL(this.action.data.card);
+        this._formatAttachment(this.action.data.attachment, embed);
+        this.payload.addEmbed(embed);
+    }
+
+    // How to Trigger?
+    async addBoardsPinnedToMember(){
+
+    }
```

Comment:
There are ~16 of these actions which I don't know how to trigger, and therefore cannot view the payload. It is possible that these actions do not trigger webhooks, however I can't be sure of that as the Trello documentation does not make clear which actions cause webhooks and which do not.

If the comment is "How to Trigger?", I do not know how to trigger said webhook. If the comment is "Won't Trigger?" then I believe I know what _should_ trigger the action however no webhook was dispatched.

If any of you guys have information about these actions let me know.